unified-runtime/source/adapters/level_zero/v2/context.cpp

```diff
@@ -67,14 +67,17 @@ ur_context_handle_t_::ur_context_handle_t_(ze_context_handle_t hContext,
                                  v2::EVENT_FLAGS_PROFILING_ENABLED)),
       p2pAccessDevices(populateP2PDevices(
           phDevices[0]->Platform->getNumDevices(), this->hDevices)),
-      defaultUSMPool(this, nullptr) {}
+      defaultUSMPool(this, nullptr),
+      asyncPool(this, nullptr) {}
 
 ur_result_t ur_context_handle_t_::retain() {
   RefCount.increment();
   return UR_RESULT_SUCCESS;
 }
 
 ur_result_t ur_context_handle_t_::release() {
+  asyncPool.cleanupPools();
```

Comment:
Shouldn't this be after decrementAndTest?

Reply:
Good catch, thanks